internal/core/app_build.go

```diff
@@ -176,6 +176,15 @@ func (op *buildOperation) Do(ctx context.Context, log hclog.Logger, app *App, _
 				log.Error("error calling dynamic func", "error", err)
 				return nil, err
 			}
+		} else {
+			if ra != nil && ra.AccessInfoFunc() == nil {
+				return nil, status.Error(codes.Internal, "The plugin requested does not "+
+					"define an AccessInfoFunc() in its Registry plugin. This is an internal "+
+					"error and should be reported to the author of the plugin.")
+			} else {
+				return nil, status.Error(codes.Internal, "The given build operation value "+
+					"is not a component.RegistryAccess type.")
+			}
```

Comment:
Pretty sure this is wrong because it will mean that any plugin that doesn't implement AccessInfoFunc fails to build, which isn't what you want. Basically you should try to run the build anyway.